src/Model/ROI.py

```diff
@@ -856,3 +860,110 @@ def renumber_roi_number(sequence):
             item.add_new(Tag("ReferencedROINumber"), "IS", str(roi_number))
         roi_number += 1
     return sequence
+
+
+def roi_to_geometry(dict_rois_contours):
+    """
+    Convert ROI contour data in each image slice to a geometry object
+    :param dict_rois_contours: A dictionary with key-value pair {slice-id: contour sequence}
+    :return: A dictionary with key-value pair {slice-id: Polygon/Multipolygon Object}
+    """
+    roi_contour_sequence_geometry = {}
+
+    for slice_id, contour_sequence in dict_rois_contours.items():
+        multi = bool(len(contour_sequence) - 1)
+
+        # Generate geometry. Handling exceptions for invalid contour data, multiple polygons
+        if not multi:
+            contour_data = contour_sequence[0]
+            try:
+                polygon = Polygon(contour_data)
+                roi_contour_sequence_geometry[slice_id] = polygon
+            except (TopologicalError, ValueError):
+                polygon = Polygon()
+                roi_contour_sequence_geometry[slice_id] = polygon
+        else:
+            polygon_list = []
+            for contour_data in contour_sequence:
+                try:
+                    sub_polygon = Polygon(contour_data)
+                    polygon_list.append(sub_polygon)
+                except (TopologicalError, ValueError):
+                    sub_polygon = Polygon()
```

Comment:
what does `sub_polygon = Polygon()` do? does this create an empty Polygon?

Reply:
Yes it is used to create an empty polygon. But this code is outdated.